portal/content/docs/contributing/architecture.md

```diff
@@ -0,0 +1,357 @@
+- Vega
+
+#### Components
+
+The client uses a component-based architecture. Each component is responsible for rendering a specific part of the user interface. The components are nested and have different level of abstraction. Most of high-level components uses `Zustand` for shared state management while low-level components are stateless or use local `React.useState`.
+
+Components that are required to have access to the server use the `Client` class singletone to access the server. The `Client` class is a wrapper around the `fetch` function that is used to send requests to the API. The `Client` class is responsible for providing client-side types information and error handling. Components get the `client` singleton instance as a prop as a part of dependency injection pattern.
+
+##### Application
+
+> Application USE `Zustand` for shared state management
+
+The `components/Application` component is the root component of the application. It is responsible for rendering the main layout of the application and managing the file browser, the routing and controllers. The underlaying component are described in the sections belows and here is an examplar structure of the `Application` component for a tabular data source:
+
+![Layout](./assets/layout.png)
+
+##### Controllers
+
+> Controllers usually USE `Zustand` for shared state management
+
+In the `components/controllers` folder there are components that are responsible for rendering a full data source representation including all the underlaying editors, views, and parts. Controllers are split by the file type e.g.:
+
+- table
+- text
+- chart
+- etc
+
+##### Editors
+
+> Editors usually USE `Zustand` for shared state management
+
+In the `components/Editors` folder there are components that are responsible for editing different data and metadata sources. It includes editors for tables, texts, Data Package metadata entities, and more.
+
+##### Views
+
+> Views usually DON'T USE `Zustand` for shared state management
+
+In the `components/Views` folder there are components that are responsible for rendering different data and metadata sources without an ability to edit the source. The most promenent example of a view is a `Report` that takes a [Frictionless Validation Report](https://framework.frictionlessdata.io/docs/framework/report.html) as one of the props and renders it to a visual representation.
+
+##### Parts
+
+> Parts usually DON'T USE `Zustand` for shared state management
+
+In the `components/Parts` folder there are components that are used in multiple places in the application. They are usually low-level components such as dialogs or buttons that are used to build more complex components. Parts are basically a application-specific library of components that mostly based on `MUI` primitives.
+
+#### Testing
+
+A testing system hasn't been set up yet. Here is a list of issue for hadnling it:
+
+- [Bootstrap unit testing for client](https://github.com/okfn/opendataeditor/issues/130)
```

Comment:
Nice touch! :100: We need to remember to update the docs as we address these tickets.